pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -71,8 +73,13 @@ func (r *DNSPolicyReconciler) reconcileGatewayHealthChecks(ctx context.Context,
 		// Keep a copy of the DNSRecord to check if it needs to be updated after
 		// reconciling the health checks
 		dnsRecordOriginal := mh.DnsRecord.DeepCopy()
+		var managedzone *v1alpha1.ManagedZone
+		for _, host := range gatewayAccessor.GetHosts() {
```

Comment:
this looks odd, looks ass though we will only get the last zone as the zone is reassigned with each iteration and then used outside of the loop. Also is there no error returned from GetManagedZoneForHost? Do we perhaps want to do a GetZoneForDNSRecord? as I recall DNSRecord references the zone

Reply:
Yes your right that's my mistake i messed up my looping and forgot my error.  Yes lets go with the GetZoneForDNSRecord instead.